python/version_details/version_details.py

```diff
@@ -466,11 +467,25 @@ def download_note_attachments(self, note_id):
 
         :param int note_id: The Note entity id.
         """
-        attachments = self.get_note_attachments(note_id)
-
-        for attachment in attachments:
-            uid = self._data_retriever.request_attachment(attachment)
-            self._attachment_uids[uid] = note_id
+        # We're going to query the list of attachments live, because don't
+        # know if the cached data for the activity stream is up to date. The
+        # reason that might be the case is that a new attachment doesn't
+        # trigger a new activity event, so the cache doesn't know it's out
+        # of date in that regard. It would be great to find a better solution
+        # than not trusting the cache.
+        self._attachment_query_uids[self._data_retriever.execute_find(
```

Comment:
My preference would be to break this up into 2 statements for readability. 13 line assignment is a bit long :P